torch/futures/__init__.py

```diff
@@ -23,6 +23,25 @@ class Future(torch._C.Future, Generic[T], metaclass=_PyFutureMeta):
     also exposes a set of APIs to add callback functions and set results.
     """
 
+    def __init__(self, devices: Optional[List[Union[int, str, torch.device]]] = None):
+        r"""
+        Create a ``Future`` that is allowed to use (a subset of) the given
+        devices. All devices must be CUDA devices (if PyTorch supports CUDA).
+        This is needed to ensure proper CUDA stream synchronization. The child
+        futures, returned by the ``then`` method, will inherit these devices.
+        """
+        if devices is None:
+            devices = []
+        device_indices = []
+        for d in devices:
+            d = torch.device(d)
+            if d.type == "cpu":
+                continue
```

Comment:
The doc says all devices must be CUDA, but we actually allow CPU?

Reply:
Though we ignored it.

Reply:
Yeah this is somewhat inconsistent. My thinking was that there is no reason to pass in a CPU device, because they're "implicit" (we'll always "synchronize" with the CPU, it's automatic), hence I didn't want to mention them in the docs to avoid creating confusion. However I didn't want to be as strict in the code to allow for some leeway.

I'm happy to address this by making the code stricter and thus making it match the docs, if that's ok?

Reply:
I am OK with either way :)